client/src/Tasks/taskRow.hooks.ts

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect, useRef, useMemo } from 'react';
+
+import { Task } from '../common/types';
+import { FILTER, TASK_COMPLETED } from './tasks.config';
+import * as TasksApi from './tasks.api';
+
+interface TaskRowState {
+    isVisible: boolean;
+    filteredLogs: Logs[];
+    setLogFilter: React.Dispatch<React.SetStateAction<FILTER>>;
+    completed: boolean;
+    expanded: boolean;
+    setExpanded: React.Dispatch<React.SetStateAction<boolean>>;
+    taskRef: React.RefObject<HTMLDivElement>;
+}
+
+export interface Logs {
+    log: string;
+    type: string;
+}
+
+export const useTaskRow = (task: Task, routeId: number): TaskRowState => {
+    const showLogs = task.id === routeId;
+    const [expanded, setExpanded] = useState<boolean>(
+        routeId ? showLogs : false,
+    );
+    const [isVisible, setVisible] = useState<boolean>(false);
+
+    const [logs, setLogs] = useState<Logs[]>([]);
+    const [logFilter, setLogFilter] = useState<FILTER>(FILTER.ALL);
+
+    const completed = task.completed || task.progress >= TASK_COMPLETED;
```

Comment:
`TASK_COMPLETED` is being compared to some numerical value here therefore we should change it name to something like `TASK_COMPLETED_TRESHOLD_VALUE`